test/base/ReferenceCounterTest.ooc

```diff
@@ -1,11 +1,69 @@
+use ooc-unit
 use ooc-base
+import threading/Thread
 
-Thing: class {
-	init: func
+TestObject: class {
+	_isAlive: Bool*
+	init: func (=_isAlive) {
+		this _isAlive@ = true
+	}
+	free: override func {
+		this _isAlive@ = false
+		super()
+	}
 }
 
-t := Thing new()
-r := ReferenceCounter new(t)
-r update(-1)
+ReferenceCounterTest: class extends Fixture {
+	init: func {
+		super("ReferenceCounter")
+		this add("single thread", func {
+			isAlive: Bool
+			object := TestObject new(isAlive&)
+			expect(isAlive)
+			counter := ReferenceCounter new(object)
+			for (i in 0 .. 10)
+				counter increase()
+			for (i in 0 .. 10)
+				counter decrease()
+			expect(isAlive, is false)
+			counter free()
+		})
+		this add("multiple threads", This _testMultipleThreads)
+	}
+	_testMultipleThreads: static func {
+		isAlive: Bool
+		object := TestObject new(isAlive&)
+		counter := ReferenceCounter new(object)
+		numberOfThreads := 8
+		countPerThread := 500
+		threads := Thread[numberOfThreads] new()
+		expect(isAlive)
+		job := func {
+			for (i in 0 .. countPerThread) {
+				counter increase()
+				Thread yield()
+				counter decrease()
+				Thread yield()
+			}
+		}
+		counter increase()
+		for (i in 0 .. numberOfThreads) {
+			threads[i] = Thread new(|| job())
+			threads[i] start()
+		}
+		for (i in 0 .. numberOfThreads) {
+			threads[i] wait()
+			threads[i] free()
+		}
+		expect(isAlive)
+		counter decrease()
+		expect(isAlive, is false)
+		(job as Closure) dispose()
+		threads free()
+		counter free()
+	}
+}
 
-"ReferenceCounterTest [TODO: Not implemented as a fixture!]" printfln()
+test := ReferenceCounterTest new()
+test run()
+test free()
```

Comment:
We (me and @davidhesselbom ) agreed that we should not create memory leaks if they can be avoided, it makes it a lot easier to use *valgrind* on the test binary. I will try to write test like that from now on. I don't know about fixing all existing tests, though.
And I know that current implementation of Fixture doesn't really care about memory leaks, but since you are planning to rewrite it I guess it's not a problem.